### ISSUE TITLE
Tailscale 1.38.1 & Geode CPU: 'Illegal instruction' after upgrade to 1.38.1

### ISSUE THREAD

**Issue by n2j3** [tailscale/tailscale#7584]
### What is the issue?

Since latest update to 1.38.1 i cannot run tailscale/tailscaled 

```
$ tailscale
Illegal instruction
```
dmesg output (from multiple attempts to run)

```
[Wed Mar 15 09:55:12 2023] traps: tailscaled[15390] trap invalid opcode ip:8092821 sp:bffee814 error:0
[Wed Mar 15 09:55:12 2023]  in tailscaled[8048000+824000]
[Wed Mar 15 09:55:16 2023] traps: tailscaled[15392] trap invalid opcode ip:8092821 sp:bf9d75a4 error:0
[Wed Mar 15 09:55:16 2023]  in tailscaled[8048000+824000]
[Wed Mar 15 09:55:29 2023] traps: tailscale[15395] trap invalid opcode ip:8091201 sp:bf8ade34 error:0
[Wed Mar 15 09:55:29 2023]  in tailscale[8048000+506000]
[Wed Mar 15 09:56:44 2023] traps: tailscale[15407] trap invalid opcode ip:8091201 sp:bf9d8714 error:0
[Wed Mar 15 09:56:44 2023]  in tailscale[8048000+506000]
[Wed Mar 15 09:56:48 2023] traps: tailscaled[15412] trap invalid opcode ip:8092821 sp:bfe7eac4 error:0
[Wed Mar 15 09:56:48 2023]  in tailscaled[8048000+824000]
[Wed Mar 15 09:57:22 2023] traps: tailscaled[15415] trap invalid opcode ip:8092821 sp:bfde0d14 error:0
[Wed Mar 15 09:57:22 2023]  in tailscaled[8048000+824000]
[Wed Mar 15 09:57:24 2023] traps: tailscale[15416] trap invalid opcode ip:8091201 sp:bfa38b54 error:0
[Wed Mar 15 09:57:24 2023]  in tailscale[8048000+506000]
[Wed Mar 15 09:57:35 2023] traps: tailscale[15418] trap invalid opcode ip:8091201 sp:bfb3d2d4 error:0
[Wed Mar 15 09:57:35 2023]  in tailscale[8048000+506000]
[Wed Mar 15 09:58:25 2023] traps: tailscale[15426] trap invalid opcode ip:8091201 sp:bf830644 error:0
[Wed Mar 15 09:58:25 2023]  in tailscale[8048000+506000]
[Wed Mar 15 10:26:43 2023] traps: tailscale[15544] trap invalid opcode ip:8091201 sp:bff467f4 error:0
[Wed Mar 15 10:26:43 2023]  in tailscale[8048000+506000]
[Wed Mar 15 10:26:55 2023] traps: tailscale[15545] trap invalid opcode ip:8091201 sp:bfeaf104 error:0
[Wed Mar 15 10:26:55 2023]  in tailscale[8048000+506000]
[Wed Mar 15 10:29:37 2023] traps: tailscale[15588] trap invalid opcode ip:8091201 sp:bfe53444 error:0
[Wed Mar 15 10:29:37 2023]  in tailscale[8048000+506000]
[Thu Mar 16 16:04:08 2023] traps: tailscale[20204] trap invalid opcode ip:8091201 sp:bfd98c24 error:0
[Thu Mar 16 16:04:08 2023]  in tailscale[8048000+506000]
```

Version installed
 *** 1.38.1 500
        500 https://pkgs.tailscale.com/stable/ubuntu focal/main i386 Packages

### Steps to reproduce

install, run

### OS

Linux

### OS version

Debian 9.13

### Tailscale version

1.38.1

**Comment by DentonGentry** [comment]
> 500 https://pkgs.tailscale.com/stable/ubuntu focal/main i386 Packages

It is a 32 bit CPU? What kind of hardware is it?

**Comment by DentonGentry** [comment]
We did a quick check using a Debian 9 VM with a kvm32 CPU, and don't run into illegal instruction errors.

If you can tell us about:
- the hardware Tailscale is running on
- the OS installed on that hardware (we note Ubuntu Focal packages being installed but the bug description says Debian 9)
- the prior working version of Tailscale

We can look into this.

**Comment by n2j3** [comment]
Last version that worked 1.36.1

I understand the hardware it runs is very niche (linutop2 , headless, 1GB flash drive) but here goes
OS: Antix 17.4
```
System:    Host: Kernel: 4.9.160-antix.2-486-smp i586 (32 bit) Console: tty 7
           Distro: antiX-17.4.1_386-net Helen Keller 28 March 2019
Machine:   Device: desktop System: First product: ION603 v: VER 2.X
           Mobo: First model: ION603 v: PCB 2.X BIOS: Phoenix v: 6.00 PG date: 11/08/2007
CPU:       Single core Geode Integrated by AMD PCS (-UP-) cache: 128 KB speed: 498 MHz (max)
Graphics:  Card: Failed to Detect Video Card!
           Display Server: N/A driver: N/A tty size: 120x29 Advanced Data: N/A out of X
Audio:     Card Advanced Micro Devices [AMD] CS5536 [Geode companion] Audio driver: snd_cs5535audio
           Sound: Advanced Linux Sound Architecture v: k4.9.160-antix.2-486-smp
Network:   Card: Realtek RTL-8100/8101L/8139 PCI Fast Ethernet Adapter driver: 8139too
           IF: eth0 state: up speed: 100 Mbps duplex: full mac: 00:40:ca:a0:89:68
Drives:    HDD Total Size: 161.1GB (81.2% used)
           ID-1: /dev/sda model: 1024MB_ATA_Fla size: 1.0GB
           ID-2: USB /dev/sdc model: 1600BEV_External size: 160.0GB
Partition: ID-1: / size: 961M used: 887M (93%) fs: ext2 dev: /dev/sda1
Info:      Processes: 95 Uptime: 24 days Memory: 213.3/464.0MB Init: SysVinit runlevel: 5
```

**Comment by DentonGentry** [comment]
> CPU:       Single core Geode Integrated by AMD PCS (-UP-) cache: 128 KB speed: 498 MHz (max)

That will be... harder. I have no such hardware and don't have a way to create a VM to pretend to be such hardware.
It is likely that we broke this inadvertently, possibly changing the compilation options for x86-32.

One thing which would be useful: we build static binaries for Geode specifically, in https://pkgs.tailscale.com/stable/#static
We do this for embedded systems using Geode processors, which frequently don't use systemd so need to have their own start/stop scripts. If you download the Geode static binaries, you can copy them into place using:
```
sudo cp tailscale /usr/bin/tailscale
sudo cp tailscaled /usr/sbin/tailscaled
```

Do those binaries work?

**Comment by n2j3** [comment]
geode binary worked as expected. Thank you!

**Comment by mikecarlton** [comment]
For anyone else who stumbles upon this -- this also fixes tailscale 1.38.1 on the PCEngines alix2d3 (also based on the Geode processor)
